src/testModel.ts

```diff
@@ -89,26 +90,38 @@ export class TestModel extends DisposableBase {
     this._collection = collection;
     this.config = { ...playwrightInfo, workspaceFolder, configFile };
     this._useLegacyCLIDriver = playwrightInfo.version < 1.44;
-    this._playwrightTest =  this._useLegacyCLIDriver ? new PlaywrightTestCLI(this._vscode, this, collection.embedder) : new PlaywrightTestServer(this._vscode, this, collection.embedder);
+    if (this._useLegacyCLIDriver) {
+      this._playwrightTest = new PlaywrightTestCLI(this._vscode, this, collection.embedder);
+      this._traceViewerFactory = new TraceViewerFactory(collection, this.config);
+    } else {
+      this._playwrightTest = new PlaywrightTestServer(this._vscode, this, collection.embedder);
+      this._traceViewerFactory = new TraceViewerFactory(collection, this.config, this._playwrightTest);
+    }
     this.tag = new this._vscode.TestTag(this.config.configFile);
-    this._spawnTraceViewer = new SpawnTraceViewer(this._vscode, this._embedder.envProvider, this.config);
 
     this._disposables = [
       this._embedder.settingsModel.showTrace.onChange(() => this._closeTraceViewerIfNeeded()),
```

Comment:
For the embedded version, I expect this toggle to show the trace viewer upon enabling.

Reply:
Addressed in eaf9074019931773ad149d158711f9184846c4db